docs/how-to-call-multiple-contracts.md

```diff
@@ -0,0 +1,139 @@
+---
+sidebar_position: 12
+---
+
+# Call Multiple Contracts in a Single Tx
+
+Up to now, we have only shown how to call one smart contract in a transaction. That is, only one input of the tx spends a smart contract UTXO, and the other inputs, if any, spend Pay-to-Public-Key-Hash ([P2PKH](https://learnmeabitcoin.com/guide/p2pkh)) UTXOs, which are generally NOT regarded as smart contracts.
+
+There are cases where it is desirable to spend multiple smart contract UTXOs in different inputs of a tx.
+
+The main differences from [calling a single contract](./how-to-deploy-and-call-a-contract.md#contract-call) are:
+
+1. Set `multiCall = true` in `MethodCallOptions`
+2. Each call may only return a partial/incomplete transaction, instead of a complete transaction
+3. A partial tx has to be passed as `CallTransation` in `MethodCallOptions` in subsequent calls
+4. Finally invoke `SmartContract.finalizeMultiCall(partialCallTransation: CallTransation, signer: Signer)` to sign and broadcast the complete transaction
+
+
+The following is an example of calling two [`Counter`](https://github.com/sCrypt-Inc/scrypt-ts-example/blob/94b0d7a374ec12d4415c56d6652af96fa8539753/src/contracts/counter.ts#L11) contracts at the same time:
```

Comment:
Use another example so people know how to call two different contracts, say Counter & HashPuzzle. Let's not use p2pkh as an example since it's NOT regarded as a contract in general.

Let's add Counter & HashPuzzle in https://github.com/sCrypt-Inc/scryptTS-examples after https://github.com/sCrypt-Inc/scrypt-ts/pull/285/ is merged.

Reply:
pr: https://github.com/sCrypt-Inc/scryptTS-examples/pull/61